vendors/nordic/boards/nrf52840-dk/ports/ota_pal_for_aws/ota_pal.c

```diff
@@ -0,0 +1,734 @@
+        xFlashEventGrp = xEventGroupCreate();
+
+        if( xFlashEventGrp == NULL )
+        {
+            LogInfo( ( "No memory for creation of event group " ) );
+            xStatus = OtaPalOutOfMemory;
+        }
+    }
+
+    if( xStatus == OtaPalSuccess )
+    {
+        /* Erase the required memory */
+        LogInfo( ( "Erasing the flash memory " ) );
+        xStatus = prvErasePages( otapalSECOND_BANK_START, otapalSECOND_BANK_END );
+
+        if( xStatus != OtaPalSuccess )
+        {
+            LogInfo( ( "Erasing the flash memory failed with error_code %d", xStatus ) );
+        }
+    }
+
+    return OTA_PAL_COMBINE_ERR( xStatus, 0 );
+}
+/*-----------------------------------------------------------*/
+
+OtaPalStatus_t otaPal_Abort( OtaFileContext_t * const pFileContext )
+{
+    pFileContext->pFile = ( int32_t ) NULL;
+
+    /* Delete the event group */
+    if( xFlashEventGrp != NULL )
+    {
+        vEventGroupDelete( xFlashEventGrp );
+        xFlashEventGrp = NULL;
+    }
+
+    return OtaPalSuccess;
+}
+/*-----------------------------------------------------------*/
+
+int16_t otaPal_WriteBlock( OtaFileContext_t * const pFileContext,
+                           uint32_t ulOffset,
+                           uint8_t * const pData,
+                           uint32_t ulBlockSize )
+{
+    /* We assume that the flash is already erased by this address (it should be, as we erase it in the
+     * otaPal_CreateFileForRx, but the second write to the same position can break this invariant.
+     * Anyway, the OTA procedure must not try to write to the same addresses */
+    ret_code_t xErrCode = prvWriteFlash( otapalSECOND_BANK_START + otapalDESCRIPTOR_SIZE + ulOffset, ulBlockSize, pData );
+
+    if( xErrCode == NRF_SUCCESS )
+    {
+        LogInfo( ( "Write %ul bytes from the %ul address ", ulBlockSize, ulOffset ) );
+        return ulBlockSize;
+    }
+    else
+    {
+        LogInfo( ( "Write %ul bytes from the %ul address failed with error code %d", ) ulBlockSize,
+                 ulOffset, xErrCode );
+        return -1;
+    }
+}
+/*-----------------------------------------------------------*/
+
+OtaPalStatus_t otaPal_CloseFile( OtaFileContext_t * const pFileContext )
+{
+    OtaPalMainStatus_t xStatus = OtaPalSuccess;
+    ret_code_t xError;
+
+    LogInfo( ( "Erasing the flash memory was successful" ) );
+
+    /* Write the image descriptor */
+    xError = otaPal_WriteImageDescriptor( pFileContext );
+
+    if( xError == NRF_SUCCESS )
+    {
+        /* Increment the file handler */
+        prvNextFreeFileHandle += 1;
+
+        xStatus = otaPal_CheckFileSignature( pFileContext );
+    }
+    else
+    {
+        xStatus = OtaPalSignatureCheckFailed;
+    }
+
+    return OTA_PAL_COMBINE_ERR( xStatus, 0 );
+}
+/*-----------------------------------------------------------*/
+
+
+static OtaPalMainStatus_t otaPal_CheckFileSignature( OtaFileContext_t * const pFileContext )
```

Comment:
Can you change the return type to be OtaPalStatus_t? Then adjust the return statements accordingly to use the "OTA_PAL_COMBINE_ERR" macro.

This is the signature that is expected by the OTA PAL tests:
static OtaPalStatus_t otaPal_CheckFileSignature( OtaFileContext_t * const pFileContext )

Also, when you update this, can you change the return to also return xErrCode as the sub error?

You will want to be careful to also update where this function is called so that the return value is handled as the OtaPalStatus_t type and not as OtaPalMainStatus_t.